flink/src/main/java/org/apache/zeppelin/flink/JarHelper.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zeppelin.flink;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.jar.JarOutputStream;
+
+/**
+ * This class copied from flink-scala-shell. Once the flink-0.9 is published in
+ * the maven repository, this class can be removed
+ *
+ * Provides utility services for jarring and unjarring files and directories.
+ * Note that a given instance of JarHelper is not threadsafe with respect to
+ * multiple jar operations.
+ *
+ * Copied from
+ * http://grepcode.com/file_/repo1.maven.org/maven2/org.apache.xmlbeans
+ * /xmlbeans/2.4.0/org/apache/xmlbeans/impl/common/JarHelper.java/?v=source
+ *
+ * @author Patrick Calahan <pcal@bea.com>
+ */
+public class JarHelper {
+  // ========================================================================
+  // Constants
+
+  private static final int BUFFER_SIZE = 2156;
+
+  // ========================================================================
+  // Variables
+
+  private byte[] mBuffer = new byte[BUFFER_SIZE];
+  private int mByteCount = 0;
+  private boolean mVerbose = false;
+  private String mDestJarName = "";
+
+  // ========================================================================
+  // Constructor
+
+  /**
+   * Instantiates a new JarHelper.
+   */
+  public JarHelper() {
+  }
+
+  // ========================================================================
+  // Public methods
+
+  /**
+   * Jars a given directory or single file into a JarOutputStream.
+   */
+  public void jarDir(File dirOrFile2Jar, File destJar) throws IOException {
+
+    if (dirOrFile2Jar == null || destJar == null) {
+      throw new IllegalArgumentException();
+    }
+
+    mDestJarName = destJar.getCanonicalPath();
+    FileOutputStream fout = new FileOutputStream(destJar);
```

Comment:
This code is executed through embedded scala compiler in this interpreter, interactively, when user run Flink job.
If there's an exception, it will propagated through callstack and displayed as a output.
Is it make sense?